modules/hal_nordic/nrfx/CMakeLists.txt

```diff
@@ -168,19 +168,29 @@ endif()
 
 dt_nodelabel(uicr_path NODELABEL "uicr")
 if(DEFINED uicr_path)
-  dt_prop(nfct_pins_as_gpios PATH ${uicr_path} PROPERTY "nfct-pins-as-gpios")
-  if(${nfct_pins_as_gpios})
-    zephyr_library_compile_definitions(
-      NRF_CONFIG_NFCT_PINS_AS_GPIOS
-    )
+  dt_prop(uicr_nfct_pins_as_gpios PATH ${uicr_path} PROPERTY "nfct-pins-as-gpios")
+  if(${uicr_nfct_pins_as_gpios})
+    set(nfct_pins_as_gpios True)
   endif()
-
   dt_prop(gpio_as_nreset PATH ${uicr_path} PROPERTY "gpio-as-nreset")
-  if(${gpio_as_nreset})
-    zephyr_library_compile_definitions(CONFIG_GPIO_AS_PINRESET)
```

Comment:
nRESET pin configuration is only available in the nRF52 Series and is always done through UICR, so I think this could be left here.